### ISSUE TITLE
Display info on feature mismatch in classifier and current image

### ISSUE THREAD

**Issue by fdsteffen** [fractal-napari-plugins-collection/napari-feature-classifier#50]
If features that were used to train a classifier are not present in the current image features the classifier currently throws an error in the console. We might instead consider to catch this error gracefully and exposing the missing features through a napari info message with the hint to include the features in the current image set. @jluethi 

*Sidenote*: Vice-versa, if features are present in image dataset but not in the classifier they are simply ignored without throwing an error.

**Comment by jluethi** [comment]
> If features that were used to train a classifier are not present in the current image features the classifier currently throws an error in the console. We might instead consider to catch this error gracefully and exposing the missing features through a napari info message with the hint to include the features in the current image set. @jluethi

Agreed, we should provide a good warning here and fail gracefully. But we have to fail, as the classifier cannot work without getting the correct features.

> Sidenote: Vice-versa, if features are present in image dataset but not in the classifier they are simply ignored without throwing an error.

That is by design. The core part: A user doesn't need to select all features to train the classifier on => the classifier needs to be able to ignore additional features.
See also discussion on not allowing the user to change feature selection later https://github.com/fractal-napari-plugins-collection/napari-feature-classifier/issues/5 and showing feature selection more explicitly https://github.com/fractal-napari-plugins-collection/napari-feature-classifier/issues/8

**Comment by fdsteffen** [comment]
> Agreed, we should provide a good warning here and fail gracefully. But we have to fail, as the classifier cannot work without getting the correct features.

Absolutely, full agree!

> That is by design.

Thanks for the pointer to previous discussion that I overlooked. Also completely agree, ignoring additional features is the expected behavior here, that's great!
